Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -64,6 +66,44 @@ public FreezeOnMatchCustomization(Type targetType)
         }
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="FreezeOnMatchCustomization"/> class.
+        /// </summary>
+        /// <param name="parameter">
+        /// The <see cref="ParameterInfo"/> used to freeze specimens decorated with advanced
+        /// attributes (e.g. <see cref="StringLengthAttribute"/>).
+        /// </param>
+        /// <param name="matcher">
+        /// The <see cref="IRequestSpecification"/> used to match the requests
+        /// that will be satisfied by the frozen specimen.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="parameter"/> or <paramref name="matcher"/> is null.
+        /// </exception>
+        public FreezeOnMatchCustomization(
+            ParameterInfo parameter,
```

Comment:
> Sorry to be a bit negative, but I think this needs to go in a slightly different direction.

That's totally fine. I found a *potential* fix, we just need to give it a proper shape ;)

I reviewed the discussion you mentioned, thanks for sharing that with me. That is a possible solution but if I understand correctly, the `TargetType` property and both existing constructors (?) in the `FreezeOnMatchCustomization` class will need to be obsoleted.

I'm still validating other possible solutions. As a POC, I created a new internal freezing [customization](https://github.com/sergeyshushlyapin/AutoFixture/blob/freezing-behavior-internal/Src/AutoFixture.xUnit.net2/FreezeOnParameterMatchCustomization.cs). It allows not to change the existing deep-seated API but it brings some other challenges with [testing internals](http://blog.ploeh.dk/2015/09/22/unit-testing-internals/).